shared/run_diff.go

```diff
@@ -104,22 +104,28 @@ func GetResultsDiff(before map[string][]int, after map[string][]int, filter Diff
 				if !filter.Deleted {
 					continue
 				}
-				diff[test] = []int{resultsBefore[1], resultsBefore[1]}
+				diff[test] = []int{resultsBefore[0], resultsBefore[1] - resultsBefore[0], -resultsBefore[1]}
 			} else {
 				if !filter.Changed && !filter.Unchanged {
 					continue
 				}
-				passDiff := abs(resultsBefore[0] - resultsAfter[0])
+				improved, regressed, delta := 0, 0, resultsBefore[0] - resultsAfter[0]
+				if delta < 0 {
```

Comment:
Well, actually this issue causes the aggregation issues we're seeing (many of) below. Ultimately, we can't compute these numbers on the client (because by that stage the summary numbers have already been squashed). We'll need to append the `diff` arrays to the `search` api.

Reply:
Now I'm a bit confused about where this data is coming from. /api/diff doesn't show up in devtools at all, how does this fit together?

In any event, it doesn't sound quite right that the aggregation bugs are due to this if-else branch at bottom. If we have the [newly-passing, newly-failing, total-delta] array of numbers for each test, then the aggregate array for some directory is just the "vector sum" of all of the test-level arrays.

What is the place where numbers are squashed in a way that isn't the "vector sum"?